cpp/src/gandiva/function_registry_string.cc

```diff
@@ -100,6 +100,9 @@ std::vector<NativeFunction> GetStringFunctionRegistry() {
                      kResultNullIfNull, "gdv_fn_like_utf8_utf8",
                      NativeFunction::kNeedsFunctionHolder),
 
+      NativeFunction("is_date", {}, DataTypeVector{utf8()}, boolean(), kResultNullIfNull,
```

Comment:
is_date should also be able to take a pattern as input for checking

Reply:
@projjal are you sure? Normally, the ISDATE function does not receive a pattern as input. Please, can you confirm it?
https://docs.oracle.com/cd/B28359_01/olap.111/b28126/dml_functions_1106.htm#OLADM567
https://docs.dremio.com/sql-reference/sql-functions/datetime-functions.html
https://docs.snowflake.com/en/sql-reference/functions/is_date-value.html